branchLayout/impl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutWriter.java

```diff
@@ -0,0 +1,55 @@
+package com.virtuslab.branchlayout.impl.readwrite;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutEntry;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.api.readwrite.IndentSpec;
+
+@CustomLog
+@RequiredArgsConstructor
+public class BranchLayoutWriter implements IBranchLayoutWriter {
+  @Override
+  public void write(OutputStream outputStream, BranchLayout branchLayout, List<String> oldFileLines)
```

Comment:
Hmmm `oldFileLines` are just needed to derive indent spec... maybe just pass an `IndentSpec` as an argument? 🤔

Reply:
This would require putting MacheteFileUtils into branchlayout.api, is this valid?

Reply:
Hmmm why specifically... which method from `MacheteFileUtils` would be required in `branchlayout.api`?

Reply:
Oh sorry, got the class name wrong. I meant `BranchLayoutFileUtils.deriveIndentSpec`. So `BranchLayoutFileUtils` will need to be moved into `branchlayout.api`

Reply:
`deriveIndentSpec` an actually be moved into `BranchLayoutWriter` (as an instance, not static method), and exposed in `IBranchLayoutWriter` 🤔